internal/services/manager_test.go

```diff
@@ -115,24 +114,6 @@ func TestAccessAuthorization(t *testing.T) {
 	require.NoError(t, err, "Teardown: could not close the client connection")
 }
 
-// requireEqualServices asserts that the grpc services were registered as expected.
-//
-// This is needed because the order of the methods and the services is not guaranteed.
-func requireEqualServices(t *testing.T, want, got map[string]grpc.ServiceInfo) {
-	t.Helper()
-
-	for name, wantInfo := range want {
-		gotInfo, ok := got[name]
-		if !ok {
-			t.Error("Expected services to match, but didn't")
-			return
-		}
-		require.ElementsMatch(t, wantInfo.Methods, gotInfo.Methods, "Expected methods to match, but didn't")
-		delete(got, name)
-	}
-	require.Empty(t, got, "Expected no extra services, but got %v", got)
-}
```

Comment:
This was used because we can't ensure the ordering of maps in Go, so even though the content is the same, the generated golden file could have the elements in a different order. Does the new compare function handle this issue?

Reply:
The compare function doesn't handle it, but the `json.Marshal` does it for us. Quoting https://pkg.go.dev/encoding/json#Marshal:

> The map keys are sorted and used as JSON object keys by applying the following rules [...]

Reply:
Eh, we're using `yaml.Marshal` not `json.Marshal`, but I think it behaves the same